WebRoot/js/zimbraAdmin/common/ZaTree.js

```diff
@@ -171,10 +180,35 @@ function(treeItem, skipNotify, kbNavEvent, noFocus) {
 
 	//this._expandUp(treeItem);
 	if (treeItem._setSelected(true, noFocus) && !skipNotify) {
-    	this._notifyListeners(DwtEvent.SELECTION, [treeItem], DwtTree.ITEM_SELECTED, null, this._selEv, kbNavEvent);
+        if (!isLicensedItem || isLicenseValid) {
+            this._notifyListeners(DwtEvent.SELECTION, [treeItem], DwtTree.ITEM_SELECTED, null, this._selEv, kbNavEvent);
+        } else {
+            ZaApp.getInstance().getCurrentController().popupErrorDialog(ZaMsg.ERROR_FEATURE_NOT_LICENSED);
+        }
 	}
 };
 
+ZaTree.prototype.setEnterSelection =
+function (treeItem, kbNavEvent) {
+    var isLicensedItem = false;
+    var isLicenseValid = false;
+    for (var i = 0; i < ZaTree.licenseCheckArray.length; i++) {
+        if (treeItem._text === ZaTree.licenseCheckArray[i].label) {
+            isLicenseValid = ZaTree.licenseCheckArray[i].licenseCheckFuncion.call();
+            isLicensedItem = true;
+            break;
+        }
+    }
+    if (!treeItem) {
+        return;
+    }
```

Comment:
This function overrides `DwtTree.prototype.setEnterSelection` if my understanding is correct.
For readability, the process defined in `DwtTree.prototype.setEnterSelection` should run first and then additional process should run.
```suggestion
ZaTree.prototype.setEnterSelection =
function (treeItem, kbNavEvent) {
    if (!treeItem) {
        return;
    }
    var isLicensedItem = false;
    var isLicenseValid = false;
    for (var i = 0; i < ZaTree.licenseCheckArray.length; i++) {
        if (treeItem._text === ZaTree.licenseCheckArray[i].label) {
            isLicenseValid = ZaTree.licenseCheckArray[i].licenseCheckFuncion.call();
            isLicensedItem = true;
            break;
        }
    }
```
i.e. moving the following code to the first part of this function.
```
    if (!treeItem) {
        return;
    }
```

Reply:
Done